Source/objects.cpp

```diff
@@ -612,52 +612,48 @@ void AddBookLever(int lx1, int ly1, int lx2, int ly2, int x1, int y1, int x2, in
 
 void InitRndBarrels()
 {
-	int v0;        // ebp
-	int v1;        // esi
-	int v2;        // edi
-	int v3;        // eax
-	BOOLEAN v4;    // ebx
-	int v5;        // edx
-	int v6;        // eax
-	int v7;        // eax
-	signed int v8; // [esp+4h] [ebp-Ch]
-	signed int v9; // [esp+8h] [ebp-8h]
-	int v10;       // [esp+Ch] [ebp-4h]
-
-	v10 = 0;
-	v0 = random(143, 5) + 3;
-	if (v0 > 0) {
+	int barrel_group_cnt;
+	int x, y;
+	int barrel_type;
+	BOOL is_location_valid;
+	int prob_to_stop_gen;
+	int direction;
+	int try_cnt;
+	int barrels_in_group_cnt;
+	int i;
+
+	barrel_group_cnt = random(143, 5) + 3;
+	for (i = 0; i < barrel_group_cnt; i++) {
 		do {
-			do {
-				v1 = random(143, 80) + 16;
-				v2 = random(143, 80) + 16;
-			} while (!RndLocOk(v1, v2));
-			v3 = random(143, 4);
-			AddObject(OBJ_BARRELEX - (v3 != 0), v1, v2);
-			v4 = 1;
-			v5 = 0;
-			v9 = 1;
-			while (!random(143, v5) && v4) {
-				v8 = 0;
-				v4 = 0;
-				do {
-					if (v8 >= 3)
-						break;
-					v6 = random(143, 8);
-					v1 += bxadd[v6];
-					v2 += byadd[v6];
-					++v8;
-					v4 = RndLocOk(v1, v2);
-				} while (!v4);
-				if (v4) {
-					v7 = random(143, 5);
-					AddObject(OBJ_BARRELEX - (v7 != 0), v1, v2);
-					++v9;
-				}
-				v5 = v9 >> 1;
+			x = random(143, 80) + 16;
+			y = random(143, 80) + 16;
+		} while (!RndLocOk(x, y));
+		barrel_type = random(143, 4) ? OBJ_BARREL : OBJ_BARRELEX;
+		AddObject(barrel_type, x, y);
+		is_location_valid = 1;
+		prob_to_stop_gen = 0;
+		barrels_in_group_cnt = 1;
+		while (random(143, prob_to_stop_gen) == 0 && is_location_valid) {
+			try_cnt = 0;
+			is_location_valid = 0;
+			while (TRUE) {
+				if (try_cnt >= 3)
+					break;
+				direction = random(143, 8);
+				x += bxadd[direction];
+				y += byadd[direction];
+				is_location_valid = RndLocOk(x, y);
+				try_cnt++;
+				if (is_location_valid)
+					break;
 			}
-			++v10;
-		} while (v10 < v0);
+			if (is_location_valid) {
+				barrel_type = random(143, 5) ? OBJ_BARREL : OBJ_BARRELEX;
+				AddObject(barrel_type, x, y);
+				++barrels_in_group_cnt;

```

Comment:
```suggestion
				barrels_in_group_cnt++;
```

Post-increment is preferred, as per the [style guide](https://github.com/diasurgical/devilution/wiki/Code-Style#increments-and-decrements).